modular_skyrat/modules/microfusion/code/microfusion_cell.dm

```diff
@@ -0,0 +1,189 @@
+	if(!attachments.len)
+		to_chat(user, span_danger("There are no attachments to remove!"))
+		return
+	remove_attachments()
+	playsound(src, 'sound/items/screwdriver.ogg', 70, TRUE)
+	to_chat(user, span_notice("You remove the upgrades from [src]."))
+
+/obj/item/stock_parts/cell/microfusion/process(delta_time)
+	for(var/obj/item/microfusion_cell_attachment/microfusion_cell_attachment as anything in attachments)
+		microfusion_cell_attachment.process_attachment(src, delta_time)
+
+/obj/item/stock_parts/cell/microfusion/examine(mob/user)
+	. = ..()
+	. += span_notice("It can hold [max_attachments] attachment(s).")
+	if(attachments.len)
+		for(var/obj/item/microfusion_cell_attachment/microfusion_cell_attachment as anything in attachments)
+			. += span_notice("It has a [microfusion_cell_attachment.name] installed.")
+		. += span_notice("Use a <b>screwdriver</b> to remove the upgrades.")
+
+/obj/item/stock_parts/cell/microfusion/proc/add_attachment(obj/item/microfusion_cell_attachment/microfusion_cell_attachment, mob/living/user)
+	if(attachments.len >= max_attachments)
+		to_chat(user, span_warning("[src] cannot fit any more attachments!"))
+		return FALSE
+	if(is_type_in_list(microfusion_cell_attachment, attachments))
+		to_chat(user, span_warning("[src] already has [microfusion_cell_attachment] installed!"))
+		return FALSE
+	attachments += microfusion_cell_attachment
+	microfusion_cell_attachment.forceMove(src)
+	microfusion_cell_attachment.add_attachment(src)
+	to_chat(user, span_notice("You successfully install [microfusion_cell_attachment] onto [src]!"))
+	playsound(src, 'sound/effects/structure_stress/pop2.ogg', 70, TRUE)
+	update_appearance()
+	return TRUE
+
+/obj/item/stock_parts/cell/microfusion/proc/remove_attachments()
+	for(var/obj/item/microfusion_cell_attachment/microfusion_cell_attachment in attachments)
+		microfusion_cell_attachment.remove_attachment(src)
+		microfusion_cell_attachment.forceMove(get_turf(src))
+		attachments -= microfusion_cell_attachment
+	update_appearance()
+
+/datum/crafting_recipe/makeshift/microfusion_cell
+	name = "Makeshift Microfusion Cell"
+	tool_behaviors = list(TOOL_SCREWDRIVER, TOOL_WIRECUTTER, TOOL_WELDER)
+	result = /obj/item/stock_parts/cell/microfusion/makeshift
+	reqs = list(/obj/item/trash/can = 1,
+				/obj/item/stack/sheet/iron = 1,
+				/obj/item/stack/cable_coil = 1)
+	time = 12 SECONDS
+	category = CAT_MISC
+
+//WHY WOULD YOU MAKE THIS?
+/obj/item/stock_parts/cell/microfusion/makeshift
+	name = "makeshift microfusion cell"
+	desc = "An.. apparatus, comprised of an everyday aluminum can with several civilian-grade batteries tightly packed together and plugged in. This vaguely resembles a microfusion cell, if the user tilts their head to a precise fifty degree angle. While the effects on enemy combatants may be dubious, it will certainly do incredible damage to the gun's warranty. What the hell were you thinking when you came up with this?"
```

Comment:
```suggestion
	desc = "An... Apparatus, comprised of an everyday aluminum can with several civilian-grade batteries tightly packed together and plugged in. This vaguely resembles a microfusion cell, if you tilt your head to a precise fifty degree angle. While the effects on enemy combatants may be dubious, it will certainly do incredible damage to the gun's warranty. What the hell were you thinking when you came up with this?"
```